sdk/storage/Azure.Storage.Files.Shares/src/ShareLeaseClient.cs

```diff
@@ -79,15 +96,43 @@ public ShareLeaseClient(ShareFileClient client, string leaseId = null)
             LeaseId = leaseId ?? CreateUniqueLeaseId();
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ShareLeaseClient"/> class.
+        /// </summary>
+        /// <param name="client">
+        /// A <see cref="ShareClient"/> representing the share being leased.
+        /// </param>
+        /// <param name="leaseId">
+        /// An optional lease ID.  If no lease ID is provided, a random lease
+        /// ID will be created.
+        /// </param>
+        public ShareLeaseClient(ShareClient client, string leaseId = null)
+        {
+            _share = client ?? throw Errors.ArgumentNull(nameof(client));
+            LeaseId = leaseId ?? CreateUniqueLeaseId();
+        }
+
         /// <summary>
         /// Gets a unique lease ID.
         /// </summary>
         /// <returns>A unique lease ID.</returns>
         private static string CreateUniqueLeaseId() => Guid.NewGuid().ToString();
 
+        /// <summary>
+        /// Ensure either the File or Share is present.
+        /// </summary>
+        private void EnsureClient()
+        {
+            if (FileClient == null && ShareClient == null)
+            {
+                // This can only happen if someone's not being careful while mocking
+                throw ShareErrors.FileOrShareMissing(nameof(ShareLeaseClient), nameof(FileClient), nameof(ShareClient));
+            }
```

Comment:
Should we validate that only one client is present ?

On the other hand shouldn't we extract common pieces somewhere and have ShareLeaseClient and FileLeaseClient?
 - it seems we branch every piece of internal logic depending on whether it's file or share (so volume of code won't change significantly if we do this.
 - some parameters are not functional for file, so we could make it stricter that way.

Reply:
This is pattern matched from [BlobLeaseClient](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/src/BlobLeaseClient.cs), I think it is important that we keep the SDKS internally consistent.